predicate.go

```diff
@@ -9,6 +10,15 @@ type predicate interface {
 	Match(root string, path string) (bool, error)
 }
 
+type PredicateError struct {
+	errType    error
+	errMessage string
+}
+
+func (p PredicateError) Error() string {
+	return fmt.Sprint(fmt.Errorf("%q: %s", p.errType, p.errMessage))
```

Comment:
Curious - why use `fmt.Errorf` here?  Would `fmt.Sprintf` be sufficient by itself?